executor/join.go

```diff
@@ -512,26 +532,31 @@ func (e *HashJoinExec) fetchInnerAndBuildHashTable(ctx context.Context) {
 		}
 		close(e.innerFinished)
 	}()
-
-	if err := e.fetchInnerRows(ctx); err != nil {
-		e.innerFinished <- errors.Trace(err)
-		return
-	}
+	// innerResultCh transfer inner result chunk from inner fetch to build hash table.
+	innerResultCh := make(chan *chunk.Chunk, e.concurrency)
+	doneCh := make(chan struct{})
+	go e.fetchInnerRows(ctx, innerResultCh, doneCh)
 
 	if e.finished.Load().(bool) {
 		return
 	}
-
-	if err := e.buildHashTableForList(); err != nil {
+	// TODO: Parallel build hash table. Currently not support because `mvmap` is not thread-safe.
```

Comment:
maybe this TODO can be removed? @XuHuaiyu how do you think?

Reply:
I think it's both ok to keep it or remove it.